physical/consul.go

```diff
@@ -513,7 +513,13 @@ shutdown:
 // checkID returns the ID used for a Consul Check.  Assume at least a read
 // lock is held.
 func (c *ConsulBackend) checkID() string {
-	return "vault-sealed-check"
+	return fmt.Sprintf("vault-sealed-check:%s:%d", c.advertiseHost, c.advertisePort)
```

Comment:
Ah, got it. If service IDs are made unique to the cluster, then it makes a lot of sense to do what you suggested. Will do.